File: app/src/main/java/com/skydoves/waterdays/presenters/SetGoalPresenter.java
package com.skydoves.waterdays.presenters;

import android.content.Context;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;

import com.skydoves.waterdays.WDApplication;
import com.skydoves.waterdays.compose.BasePresenter;
import com.skydoves.waterdays.persistence.preference.PreferenceKeys;
import com.skydoves.waterdays.persistence.preference.PreferenceManager;
import com.skydoves.waterdays.viewTypes.SetGoalActivityView;

import javax.inject.Inject;

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

public class SetGoalPresenter extends BasePresenter<SetGoalActivityView> {

    protected @Inject PreferenceManager preferenceManager;

    @Override
    protected void onCreate(@NonNull Context context, @Nullable Bundle savedInstanceState) {
        super.onCreate(context, savedInstanceState);
        WDApplication.getComponent().inject(this);
    }

    public void onClickSetGoal(String value) {
        if(checkGoalValid(value)) {
            setWaterGoal(value);
            baseView.onSetSuccess();
            if(getIsNewbe()) baseView.intentMain();
        } else
            baseView.onSetFailure();
    }

    public boolean checkGoalValid(String value) {
        if(value.equals(""))
            return false;
        else if(value.equals("0"))
            return false;
        else
            return true;
    }

    public boolean getIsNewbe() {
        return preferenceManager.getBoolean(PreferenceKeys.NEWBE.first, PreferenceKeys.NEWBE.second);
    }

    public void setOldbe() {
         preferenceManager.putBoolean(PreferenceKeys.NEWBE.first, false);
    }

    public String getWaterGoal() {
        return preferenceManager.getString(PreferenceKeys.WATER_GOAL.first, PreferenceKeys.WATER_GOAL.second);
    }

    public void setWaterGoal(String goal) {
        preferenceManager.putString(PreferenceKeys.WATER_GOAL.first, goal);
    }
}

File: app/src/main/java/com/skydoves/waterdays/presenters/SetGoalPresenter.kt
/*
 * Copyright (C) 2016 skydoves
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.skydoves.waterdays.presenters

import android.content.Context
import android.os.Bundle

import com.skydoves.waterdays.WDApplication
import com.skydoves.waterdays.compose.BasePresenter
import com.skydoves.waterdays.persistence.preference.PreferenceKeys
import com.skydoves.waterdays.persistence.preference.PreferenceManager
import com.skydoves.waterdays.viewTypes.SetGoalActivityView

import javax.inject.Inject

/**
 * Developed by skydoves on 2017-08-19.
 * Copyright (c) 2017 skydoves rights reserved.
 */

class SetGoalPresenter : BasePresenter<SetGoalActivityView>() {

  @Inject
  lateinit var preferenceManager: PreferenceManager

  override fun onCreate(context: Context, savedInstanceState: Bundle?) {
    super.onCreate(context, savedInstanceState)
    WDApplication.component.inject(this)
  }

  fun onClickSetGoal(value: String) {
    if (checkGoalValid(value)) {
      waterGoal = value
      baseView.onSetSuccess()
      if (isNewbe) baseView.intentMain()
    } else
      baseView.onSetFailure()
  }

  fun checkGoalValid(value: String): Boolean {
    when (value) {
      "" -> return false
      "0" -> return false
    }
    return true
  }

  val isNewbe: Boolean by lazy {
    preferenceManager.getBoolean(PreferenceKeys.NEWBE.first, PreferenceKeys.NEWBE.second)
  }

  fun setOldbe() = preferenceManager.putBoolean(PreferenceKeys.NEWBE.first, false)

  var waterGoal: String
    get() = preferenceManager.getString(PreferenceKeys.WATER_GOAL.first, PreferenceKeys.WATER_GOAL.second)
    set(goal) = preferenceManager.putString(PreferenceKeys.WATER_GOAL.first, goal)
}
